react/src/redux/books/actions.js

```diff
@@ -22,9 +26,66 @@ const actionCreators = {
   },
   getBook: (id) => {
     return async (dispatch) => {
-      const response = await getBookService(id);
-      if(response.ok){
-        dispatch({type: actionTypes.GET_BOOK_SUCCESS, payload:{currentBook: response.data}})
+      dispatch({type: actionTypes.GET_BOOK_LOADING});
+      const responseBook = await getBookService(id);
+      if(responseBook.ok){
+        const responseRents = await getBookRentsService(id);
+        if(responseRents.ok){
+          const rentData = responseRents.data.some(rent => rent.user.email === sessionStorage.getItem('user_session'));
+          if(rentData){
+            dispatch({
+              type: actionTypes.GET_BOOK_SUCCESS,
```

Comment:
I'd extract the creation of this action in a function, so we can reuse it

Reply:
I'd also decouple `bookState` from this action, since it starts to do many things. What do you think ?

Reply:
What do you mean by bookState?
If you are referring to bookDetailStates, how would you decouple it?